### PR TITLE
Don't push LAT releases to amazon

### PR DESCRIPTION
**Pull request by mitchhentges**
This avoids submitting LAT apps to Amazon (the API doesn't support LAT releases, and existing automation behaviour is to create an "upcoming version"), and adjusts email content to link to the unsigned apk.

## Checklist

- [x] Confirm the **acceptance criteria** is fully satisfied in the issue(s) this PR will close
- [x] Add thorough **tests** or an explanation of why it does not
- [x] Add a **CHANGELOG entry** if applicable
- [x] Add **QA labels** on the associated issue (not this PR; `qa-ready` or `qa-notneeded`)
- [x] This is tested

### CODE REVIEW

File: taskcluster/firefox_tv_taskgraph/target_tasks.py [hunk 1]
```diff
@@ -7,11 +7,27 @@
...
+        release_type = parameters["release_type"]
+
+        if release_type == "LAT":
+            def filter(task, params):
+                return tag_filter("lat", task, params)
+        else:
+            def filter(task, params):
+                return tag_filter("production", task, params)
+    else:
+        def filter(task, params):
```
Comment (mcomella) [inline taskcluster/firefox_tv_taskgraph/target_tasks.py]: nit: I don't love that this name shadows a built-in function name.

Comment (mitchhentges) [reply taskcluster/firefox_tv_taskgraph/target_tasks.py]: That's a good point, though I'm following the pattern from `fenix`.
I'm not super familiar with `taskgraph`, but so far it seems like the existing convention

File: taskcluster/ci/email/kind.yml [hunk 2]
```diff
@@ -15,13 +15,19 @@ jobs:
     worker-type: succeed
     run-on-tasks-for: [github-release]
     dependencies:
+      build: build-release
```
Comment (JohanLorenzo) [inline taskcluster/ci/email/kind.yml]: Nit: What does specifying this dependency add? As far as I understand `Sign for Github`, already depends on `build-release`, I don't think we need to depend on `build-release` explicitly.

https://github.com/mozilla-mobile/firefox-tv/blob/50e5e72e8e913750cd5d669c26002a8cd43a6678/taskcluster/ci/signing/kind.yml#L18

File: taskcluster/firefox_tv_taskgraph/email.py [hunk 3]
```diff
@@ -18,23 +18,20 @@
 @transforms.add
 def email_task(config, tasks):
     tag = config.params.get("head_tag")
+    release_type = config.params.get("release_type")
 
     for task in tasks:
         to_address = task["worker"].pop("to-address")
-        content = task["worker"].pop("content")
+        content_chooser = task["worker"].pop("content")
+        content = content_chooser["by-release-type"].get(release_type, "")
```
Comment (JohanLorenzo) [inline taskcluster/firefox_tv_taskgraph/email.py]: Nit: Use `resolve_keyed_by()`. It handles all cases for you. For instance: https://github.com/mozilla-mobile/fenix/blob/6a8f7224dc6c583f480a258ba7af796fa562e935/taskcluster/fenix_taskgraph/transforms/signing.py#L22

File: taskcluster/firefox_tv_taskgraph/target_tasks.py (hunk 1 above)
Comment (JohanLorenzo) [reply taskcluster/firefox_tv_taskgraph/target_tasks.py]: I've followed this pattern from mozilla-central too. I've never taken the time to understand why it's okay to shadow this built-in.

File: taskcluster/firefox_tv_taskgraph/target_tasks.py [hunk 4]
```diff
@@ -7,11 +7,27 @@
...
+
+
 @_target_task('default')
 def target_tasks_default(full_task_graph, parameters, graph_config):
     """Target the tasks which have indicated they should be run on this project
     via the `run_on_projects` attributes."""
-    def filter(task, params):
-        return standard_filter(task, params)
+
+    if parameters["tasks_for"] == "github-release":
```
Comment (JohanLorenzo) [inline taskcluster/firefox_tv_taskgraph/target_tasks.py]: Like Tom said, it's better to move this `if` statement at the beginning of the pipeline, in https://github.com/mozilla-mobile/fenix/pull/5564/files#diff-53e691d50a0b26d43d925c2150567e76R52. 

Can be done in a followup.

File: taskcluster/ci/signing/kind.yml [hunk 5]
```diff
@@ -9,7 +9,7 @@ transforms:
   - taskgraph.transforms.task:transforms
 
 jobs:
-  push:
+  sign:
```
Comment (JohanLorenzo) [inline taskcluster/ci/signing/kind.yml]: Followup: For clarity, it'd be nice to use a `single_dep` loader. This way, you don't have to:

1. Define 1 job per build-type
2. Handle dependencies manually.

You just have a template that you apply to every task that is loaded in this kind.

You can easily filter out based on an attribute, for instance:  https://github.com/mozilla-mobile/fenix/blob/2896b36d14362598a7f8e62e311e54db17991a24/taskcluster/ci/push-apk/kind.yml#L15

Example: https://github.com/mozilla-mobile/fenix/blob/2896b36d14362598a7f8e62e311e54db17991a24/taskcluster/fenix_taskgraph/loader/single_dep.py (this loader should move up to taskgraph, I think)

File: taskcluster/ci/push-apk/kind.yml [hunk 6]
```diff
@@ -10,6 +10,8 @@ transforms:
 
 jobs:
   push:
+    attributes:
+      release-type: production
```
Comment (JohanLorenzo) [inline taskcluster/ci/push-apk/kind.yml]: Followup: Same comment about `dep_loader`

File: taskcluster/ci/email/kind.yml (hunk 2 above)
Comment (mitchhentges) [reply taskcluster/ci/email/kind.yml]: Oh, but I have an artifact-reference to `<build/...>`. Doesn't that mean I need to depend on `build-release` explicitly?

Comment (JohanLorenzo) [reply taskcluster/ci/email/kind.yml]: Oh, that's a good question. I'm not sure for this particular case. Anyway, let's keep it for now. `kind-dependencies` will make manual `dependencies` go away, anyway.

**Comment by mitchhentges** [comment]
@mcomella this is ready to land whenever you're ready

**Pull request merged**